app/controllers/view_components_controller.rb

```diff
@@ -56,7 +58,26 @@ def set_locale
     end
   end
 
-  def prepend_application_view_paths
-    prepend_view_path Rails.root.join("app/views") if defined?(Rails.root)
+  # By default, it should use the Rails internal layout, or the app's layout if
+  # always_use_app_views is set to +true+.
+  #
+  # Returns either {} or {layout: value} depending on configuration
+  def determine_layout(layout_override = nil, always_use_app_views = false)
+    return {} unless defined?(Rails.root)
+
+    layout_declaration = {}
+
+    if !layout_override.nil?
+      # Allow component-level override, even if false (thus no layout rendered)
+      layout_declaration[:layout] = layout_override
+    elsif default_preview_layout.present?
+      layout_declaration[:layout] = default_preview_layout
+    end
+
+    if layout_declaration.key?(:layout) || !!always_use_app_views
```

Comment:
IMHO no need to double negate, it's just a boolean.

```suggestion
    if layout_declaration.key?(:layout) || always_use_app_views
```